packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,245 @@
+      revert IGachaLazyClaim.ClaimAlreadyInitialized();
+
+    // Sanity checks
+    if (claimParameters.storageProtocol != StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    address[] memory receivers = new address[](1);
+    receivers[0] = msg.sender;
+    string[] memory uris = new string[](1);
+    // TODO check on logic to reserve tokenIds for claim itemVariations
+    uint256[] memory amounts = new uint256[](claimParameters.itemVariations);
+    uint256[] memory newTokenIds = IERC1155CreatorCore(creatorContractAddress).mintExtensionNew(receivers, amounts, uris);
+
+    require(newTokenIds[0] <= type(uint112).max, "Token ID exceeds uint112 range");
+
+
+    // Create the claim
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claimParameters.storageProtocol,
+      total: 0,
+      totalMax: claimParameters.totalMax,
+      startDate: claimParameters.startDate,
+      endDate: claimParameters.endDate,
+      startingTokenId: uint112(newTokenIds[0]),
+      itemVariations: claimParameters.itemVariations,
+      location: claimParameters.location,
+      paymentReceiver: claimParameters.paymentReceiver,
+      erc20: claimParameters.erc20,
+      cost: claimParameters.cost
+    });
+    for (uint8 i; i < claimParameters.itemVariations; i++) {
+      _claimTokenIds[creatorContractAddress][newTokenIds[i]] = instanceId;
+    }
+
+    emit GachaClaimInitialized(creatorContractAddress, instanceId, msg.sender);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-updateClaim}.
+   */
+  function updateClaim(
+    address creatorContractAddress,
+    uint256 instanceId,
+    ClaimParameters memory claimParameters
+  ) external override adminRequired {
+    Claim memory claim = _getClaim(creatorContractAddress, instanceId);
+    if (claimParameters.endDate != 0 && claimParameters.startDate >= claimParameters.endDate)
+      revert IGachaLazyClaim.InvalidStartDate();
+    if (claimParameters.totalMax != claim.totalMax) revert IGachaLazyClaim.CannotChangeTotalMax();
+    if (claimParameters.startingTokenId != claim.startingTokenId) revert IGachaLazyClaim.CannotChangeStartingTokenId();
+    if (claimParameters.itemVariations != claim.itemVariations) revert IGachaLazyClaim.CannotChangeItemVariations();
+    if (claimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    if (keccak256(abi.encodePacked(claimParameters.location)) != keccak256(abi.encodePacked(claim.location))) {
+      revert IGachaLazyClaim.CannotChangeLocation();
+    }
+    if (claimParameters.paymentReceiver != claim.paymentReceiver) revert IGachaLazyClaim.CannotChangePaymentReceiver();
+    if (claimParameters.erc20 != claim.erc20) revert IGachaLazyClaim.CannotChangePaymentToken();
+
+    // Overwrite the existing values for startDate, endDate, and cost
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claim.storageProtocol,
```

Comment:
A lot of these references seeem wrong and should be refrencing claimParameters

Reply:
yeah this is intentional for now since we're constraining to only be able to update startDate, endDate and Cost for a gacha claim right now

Reply:
Even if you do that in the app, you should not constrain the smart contract, since it's immutable

Reply:
@nikki-kiga we should be able to update the storage protocol.  Right here you seem to be unable to in line 106

Reply:
This is still incorrect @nikki-kiga 

You're not updating the storageProtocol with what's specified in he claimParameters